### DISCUSSION TITLE
How can I use TabBar in custom widget

### DISCUSSION THREAD

**Discussion by goflutterjava** [hecrj/iced#1259] [Q&A]
I want to use `TabBar` in my custom widget.
How can custom widget defines callback functions like `Application` does. (ex: `Message::TabSelected`)

**Comment by hecrj** [comment]
What are you trying to achieve exactly? Let's try to avoid [The XY Problem](https://xyproblem.info/).

**Comment by goflutterjava** [comment]
@hecrj I want to seal the `TabBar` with it's business logic into a custom widget.
So I can arrange the ability in different widget, independent test, maintain, develop
